### PR TITLE
Framework: Remove middleware that resets selected site

### PR DESCRIPTION
**Pull request by seear**
Remove use of `sitesList.getSelectedSite`, since this middleware does not seem to be necessary anymore. Was added with the following comment:

> This pull request seeks to avoid an undesirable consequence of site resetting where, if the next section is not site-specific but requires a section chunk to be loaded, the site is reset but the next page isn't rendered until chunk loading has finished.

> Steps to reproduce:

> In a fresh Calypso page session, navigate to My Sites
Select a site if one isn't already selected
Click your profile image to the right of the master bar (i.e. /me)
In master: You'll see that the current My Sites view resets to All Sites while /me loads
In this branch: The current My Sites view maintains the selected site until after /me loads

> You should also confirm that the clicking the New Post button in the master bar after navigating to /me still correctly shows the site dropdown.

This PR also adds `getCurrentSiteSlug()` selector for convenience.

**To Test**
* Test case above
* Check that the help contact from behaves as master by showing the (last) selected site or primary site

### CODE REVIEW

**Comment by seear** [comment]
cc @aduth original change was yours

**Comment by aduth** [comment]
Related (duplicate?): #8281 (cc @tyxla)

There's some background in that pull request to the challenges related in removing this.

**Comment by seear** [comment]
> Related (duplicate?): #8281 (cc @tyxla)

Aha!

@tyxla you may want to take the bits of this PR that mean you don't introduce another use of [`getSelectedSite`](https://github.com/Automattic/wp-calypso/pull/8281/files#diff-bc7120f8349b381d24b34e6e3a2daeb2R73), since we are trying to [remove those](https://github.com/Automattic/wp-calypso/issues/8726).

**Comment by seear** [comment]
Having said that, may be easier to wait until yours is merged, then I'll rebase this. I do think the addition of `getCurrentSiteSlug()` in this PR will be useful—I know a couple of other places where it will tidy things up.

**Comment by tyxla** [comment]
Thanks for pinging me here, folks! I'll make sure to remove the `getSelectedSite` call soon. I'll ping you when that's ready to review again @seear 👍

**Comment by tyxla** [comment]
@seear I've addressed your suggestions in #8281, thank you! Feel free to review and test it. Also, I feel we can close this PR if we land mine.

File: client/me/help/help-contact-form/index.jsx [hunk 1]
```diff
@@ -20,13 +21,14 @@ import FormTextInput from 'components/forms/form-text-input';
...
+import { getSelectedSiteSlug } from 'state/ui/selectors';
 
 /**
  * Module variables
  */
 const sites = siteList();
 
-module.exports = React.createClass( {
+const HelpContactForm = React.createClass( {
 	displayName: 'HelpContactForm',
```
Comment (aduth) [inline client/me/help/help-contact-form/index.jsx]: Looks like this was incorporated in 5ca23c9e822ad47299ee369b415b2cf172872cab (#8281). Thanks for catching @spentay !

**Pull request closed without merging**